.github/workflows/ci.yml

```diff
@@ -0,0 +1,120 @@
+name: CI
+
+permissions:
+  contents: read
+  packages: read
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
```

Comment:
I usually go with `${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}` in my projects (can't remember why though).
So out of curiosity, any specific reason why you went with `${{ github.workflow }}-${{ github.ref }}`, and do you think I should switch to it too?

Reply:
No reason. This is a carryover from a different old project of mine. I'm very happy to figure out new best practices that I can take to my other repos.

Reply:
So I investigated it a bit. And it looks I should replace `github.sha` with `github.ref`. But the `github.event.pull_request.number` seems to be recommended as a good approach when combined with `cancel-in-progress: true` (on the line below this one). The `||` short-circuits, so the `github.ref` (currently `.ref`) is a fallback in case there's no PR involved.

On github's own docs I only saw the `github.ref` being used like done here, so it's probably fine to keep it like this.